controllers/policies/policystatus_controller.go

```diff
@@ -0,0 +1,124 @@
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getPolicyMapFromConfigMap(configMap *corev1.ConfigMap) (admission.PolicyConfigEntryMap, error) {
+	policyMap := admission.PolicyConfigEntryMap{}
+	if policies, ok := configMap.Data[constants.PolicyServerConfigPoliciesEntry]; ok {
+		if err := json.Unmarshal([]byte(policies), &policyMap); err != nil {
+			return policyMap, err
+		}
+	} else {
+		return policyMap, nil
+	}
+	return policyMap, nil
+}
+
+func SetPolicyConfigurationCondition(policyServerConfigMap *corev1.ConfigMap, policyServerDeployment *appsv1.Deployment, conditions *[]metav1.Condition) {
+	if configAnnotation, ok := policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation]; ok {
+		if configAnnotation == policyServerConfigMap.ResourceVersion {
+			apimeta.SetStatusCondition(
+				conditions,
+				metav1.Condition{
+					Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+					Status:  metav1.ConditionTrue,
+					Reason:  "ConfigurationVersionMatch",
+					Message: "Configuration for this policy is up to date",
+				},
+			)
+		} else {
+			apimeta.SetStatusCondition(
+				conditions,
+				metav1.Condition{
+					Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+					Status:  metav1.ConditionFalse,
+					Reason:  "ConfigurationVersionMismatch",
+					Message: "Configuration for this policy is not up to date",
+				},
+			)
+		}
+	} else {
+		apimeta.SetStatusCondition(
+			conditions,
+			metav1.Condition{
+				Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+				Status:  metav1.ConditionFalse,
+				Reason:  "UnknownConfigurationVersion",
+				Message: fmt.Sprintf("Configuration version annotation (%s) in deployment %s is missing", constants.PolicyServerDeploymentConfigVersionAnnotation, policyServerDeployment.GetName()),
+			},
+		)
+	}
+}
+
+func SetPolicyUniquenessCondition(ctx context.Context, apiReader client.Reader, policyServerConfigMap *corev1.ConfigMap, policyServerDeployment *appsv1.Deployment, conditions *[]metav1.Condition) {
+	replicaSets := appsv1.ReplicaSetList{}
+	if err := apiReader.List(ctx, &replicaSets, client.InNamespace(policyServerDeployment.Namespace)); err != nil {
+		return
+	}
+	podTemplateHash := ""
+	for _, replicaSet := range replicaSets.Items {
+		if replicaSet.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] == policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] {
+			podTemplateHash = replicaSet.Labels[appsv1.DefaultDeploymentUniqueLabelKey]
+			break
+		}
+	}
+	if podTemplateHash == "" {
+		return
+	}
+	pods := corev1.PodList{}
+	if err := apiReader.List(ctx, &pods, client.InNamespace(policyServerDeployment.Namespace)); err != nil {
```

Comment:
Does this get all pods for this namespace? we should check just pods for the policy server replicaset. Otherwise [this](https://github.com/kubewarden/kubewarden-controller/pull/175/files#diff-bd42f4c88fc02e50f4541d17e900e6764497a22ae5a74fd28d8b8e5c3137f60eR102 ) will fail if there are more pods in the policy server namespace